plugins/woocommerce/tests/e2e-pw/tests/merchant/products/block-editor/product-attributes-block-editor.spec.js

```diff
@@ -94,111 +95,141 @@ const test = baseTest.extend( {
+				terms: [ 'Modern', 'Classic', 'Vintage' ],
 			},
 		];
 
-		await test.step( 'go to product editor, Organization tab', async () => {
+		await test.step( 'go to product editor -> Organization tab -> Click on `Add new`', async () => {
 			await page.goto(
 				`wp-admin/post.php?post=${ product.id }&action=edit`
 			);
-			await page.getByRole( 'tab', { name: 'Organization' } ).click();
-		} );
+			await clickOnTab( 'Organization', page );
 
-		await test.step( 'add new attributes', async () => {
-			await page.getByRole( 'button', { name: 'Add new' } ).click();
-
-			await page.waitForLoadState( 'domcontentloaded' );
-
-			// Add attributes that do not exist
-			await page.getByPlaceholder( 'Search or create attribute' ).click();
-
-			// Unless we wait for the list to be visible, the attribute name will be filled too soon and the test will fail.
-			await waitForAttributeList( page );
-
-			await page
-				.getByPlaceholder( 'Search or create attribute' )
-				.fill( newAttributes[ 0 ].name );
 			await page
-				.getByText( `Create "${ newAttributes[ 0 ].name }"` )
-				.click();
+				.getByRole( 'heading', { name: 'Attributes' } )
+				.isVisible();
 
-			await page.getByPlaceholder( 'Search or create value' ).click();
-			await page
-				.getByPlaceholder( 'Search or create value' )
-				.fill( newAttributes[ 0 ].value );
-			await page
-				.getByText( `Create "${ newAttributes[ 0 ].value }"` )
-				.click();
+			await page.getByRole( 'button', { name: 'Add new' } ).click();
 
-			// Add another attribute
-			await page.getByLabel( 'Add another attribute' ).click();
 			await page
-				.getByPlaceholder( 'Search or create attribute' )
-				.nth( 1 )
-				.click();
+				.getByRole( 'heading', { name: 'Add variation options' } )
+				.isVisible();
 
-			await waitForAttributeList( page );
+			await page.waitForLoadState( 'domcontentloaded' );
+		} );
 
-			await page
-				.getByPlaceholder( 'Search or create attribute' )
-				.nth( 1 )
-				.fill( newAttributes[ 1 ].name );
-			await page
-				.getByText( `Create "${ newAttributes[ 1 ].name }"` )
-				.click();
+		await test.step( 'create local attributes with terms', async () => {
+			/*
+			 * attributeRowsLocator are the rows that contains
+			 * the Attribute combobox and the Term FormTokenField.
+			 */
+			const attributeRowsLocator = page.locator(
+				'.woocommerce-new-attribute-modal__table-row'
+			);
 
-			await page.getByPlaceholder( 'Search or create value' ).click();
-			await page
-				.getByPlaceholder( 'Search or create value' )
-				.fill( newAttributes[ 1 ].value );
-			await page
-				.getByText( `Create "${ newAttributes[ 1 ].value }"` )
-				.click();
+			/*
+			 * First, check the app loads the attributes,
+			 * based on the Spinner visibility.
+			 */
+			const spinnerLocator = attributeRowsLocator.locator(
+				'.components-spinner'
+			);
+			await spinnerLocator.waitFor( {
+				state: 'visible',
+			} );
+			await spinnerLocator.waitFor( { state: 'hidden' } );
```

Comment:
Is there another way that we could wait? I'm just wondering if there could be a case where the spinner never shows because the attributes load too quickly?

Reply:
We could rely on the async request. However, I'm inclined to believe that end-to-end (E2E) tests are closer to the UI compared to other test types. That's why I think detecting changes in the UI is good. Maybe I'm wrong.

From a technical implementation point of view, the Spinner should always be visible, no matter how short the async request could take.
But, I wonder now what happens if the app populates (performs the request) in another instance of the app. In that case, the Spinner never shows.
But (😅 ), the same thing will happen with the check of the async request since it's already done.

For now, we're fine. If we introduce changes in how the data is propagated, then probably we'll have to update the test.

Reply:
A good alternative is adding another element/text that will always be visible, such as a placeholder. The thing is, if I recall properly, we don't want to add a placeholder text in the Attribute Combobox for some design reason. Going to ask